flink-yarn/src/main/java/org/apache/flink/yarn/Utils.java

```diff
@@ -255,4 +257,33 @@ private Utils() {
 		}
 		return result;
 	}
+
+	/**
+	 * Recursive directory copy to work around FileSystem implementations that do not implement it.
+	 * @param fs
+	 * @param localPath
+	 * @param remotePath
+	 * @throws IOException
+     */
+	protected static void copyFromLocalFile(final FileSystem fs, final Path localPath, final Path remotePath) throws IOException {
+		File localFile = new File(localPath.toUri());
+		if (localFile.isDirectory()) {
+			for (File file : localFile.listFiles()) {
+				copyFromLocalFile(fs, new Path("file://" + file.getAbsolutePath()), new Path(remotePath,file.getName()));
+			}
+		} else {
+			fs.copyFromLocalFile(localPath,remotePath);
+		}
+	}
+
+	public static Path checkScheme(final Path localRsrcPath) throws IOException {
```

Comment:
Can you make this private and remove the `final` modifier? What is the current behaviour if no scheme is specified?